buildbot/cmd/buildbot/rotation.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/log"
+)
+
+// The Rotation object which knows how to dynamically find the correct build cop from
+// the file getter f it is initialized with.
+type Rotation struct {
+	f GetFile
+}
+
+// GetFile is the signature of a function that knows how to retrieve the bytes from a file
+type GetFile func() (io.ReadCloser, error)
+
+// NewRotationFromFile returns a new Rotation object which will read from file.
+func NewRotationFromFile(file string) Rotation {
```

Comment:
Also, this is only used for the tests right, so… it should only be defined there :stuck_out_tongue_winking_eye:

Reply:
I guess so - it only _happens_ to be used by the unit tests, but it's a pretty reasonable part of this overall package

Reply:
:stuck_out_tongue_closed_eyes: true, but it ain't use by anything except the test, it doesn't need to be exported :angel: 
If anybody needs this to be exported at some point, a PR can be opened :angel: 

(that's usually my take on things : whatever doesn't need to be exported *right now*, shouldn't be exported)

Reply:
kk ill try it! i spose it makes sense to expose as little as possible, that sounds like a principle i learned a while ago or something 🤔